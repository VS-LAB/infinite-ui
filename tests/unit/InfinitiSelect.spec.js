/*
 * @Author: your name
 * @Date: 2021-02-05 14:31:42
 * @LastEditTime: 2021-02-05 15:03:23
 * @LastEditors: Please set LastEditors
 * @Description: In User Settings Edit
 * @FilePath: /infinite-ui/tests/unit/InfinitiSelect.spec.js
 */
import { mount } from '@vue/test-utils'

import InfiniteSelect from '@/packages/infinite-select/src/index.vue'

describe('InfiniteSelect', () => {
  const wrapper = mount(InfiniteSelect, {
    propsData: {
      value: '123'
    }
  })

  it('toMatchSnapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('select exists', () => {
    const select = wrapper.findComponent({
      ref: 'infiniteSelect'
    })
    expect(select.exists()).toBe(true)
  })

  it('visible-change true', () => {
    const select = wrapper.findComponent({
      ref: 'infiniteSelect'
    })
    const mock = jest.fn()
    select.vm.blur = mock
    select.vm.$emit('visible-change', true)
    expect(mock).toBeCalledTimes(0)
    select.vm.$emit('visible-change', false)
    expect(mock).toBeCalledTimes(1)
  })
})
