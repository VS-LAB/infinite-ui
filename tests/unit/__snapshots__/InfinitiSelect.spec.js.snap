// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfiniteSelect toMatchSnapshot 1`] = `
<div class="el-select">
  <!---->
  <div class="el-input el-input--suffix">
    <!----><input type="text" readonly="readonly" autocomplete="off" placeholder="请选择" class="el-input__inner">
    <!----><span class="el-input__suffix"><span class="el-input__suffix-inner"><i class="el-select__caret el-input__icon el-icon-arrow-up"></i><!----><!----><!----><!----><!----></span>
    <!----></span>
    <!---->
    <!---->
  </div>
  <transition-stub name="el-zoom-in-top">
    <div class="el-select-dropdown el-popper" style="display: none;">
      <div class="el-scrollbar" style="display: none;">
        <div class="el-select-dropdown__wrap el-scrollbar__wrap el-scrollbar__wrap--hidden-default">
          <ul class="el-scrollbar__view el-select-dropdown__list"></ul>
        </div>
        <div class="el-scrollbar__bar is-horizontal">
          <div class="el-scrollbar__thumb" style="transform: translateX(0%); webkit-transform: translateX(0%);"></div>
        </div>
        <div class="el-scrollbar__bar is-vertical">
          <div class="el-scrollbar__thumb" style="transform: translateY(0%); webkit-transform: translateY(0%);"></div>
        </div>
      </div>
      <p class="el-select-dropdown__empty">
        无数据
      </p>
    </div>
  </transition-stub>
</div>
`;
